crates/jarscanner/src/lib.rs

```diff
@@ -0,0 +1,200 @@
+use crate::errors::FileNameError;
+use serde::{Deserialize, Serialize};
+
+use std::collections::HashMap;
+use std::error::Error;
+
+use std::fs::File;
+use std::io::prelude::*;
+use std::path::PathBuf;
+use zip::read::ZipArchive;
+
+pub mod errors;
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Default, Eq)]
+struct DataBlock {
```

Comment:
seems weird we need to copy this. Isn't code like this already in another project here? Can we factor so we can share this code across crates?

Reply:
Ah, yeah it is! A little bit of newbie-ness here. Will fix